Shared/Samples/Filter by definition expression or display filter/FilterByDefinitionExpressionOrDisplayFilterView.swift

```diff
@@ -0,0 +1,184 @@
+                    model.filterFeatureLayer(filterMode: newFilterMode)
+                }
+            }
+        }
+    }
+}
+
+private extension FilterByDefinitionExpressionOrDisplayFilterView {
+    /// The view model for the sample.
+    final class Model: ObservableObject {
+        /// A map with a topographic basemap.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISTopographic)
+            
+            // Initially centers the map on San Fransisco, CA, USA.
+            map.initialViewpoint = Viewpoint(
+                center: Point(latitude: 37.7762, longitude: -122.4522),
+                scale: 7e4
+            )
+            return map
+        }()
+        
+        /// A feature layer of San Fransisco 311 incidents.
+        private let featureLayer: FeatureLayer = {
+            // Creates the feature layer from a feature service URL.
+            let featureTable = ServiceFeatureTable(url: .sanFransiscoIncidentsFeatureLayer)
+            return FeatureLayer(featureTable: featureTable)
+        }()
+        
+        /// The parameters for querying the feature layer.
+        private let queryParameters = QueryParameters()
+        
+        /// A definition expression to filter for the "Tree Maintenance or Damage" type.
+        private let treesDefinitionExpression = "req_type = 'Tree Maintenance or Damage'"
+        
+        /// A display filter definition to filter for the "Tree Maintenance or Damage" type.
+        private let treesDisplayFilterDefinition: ManualDisplayFilterDefinition = {
+            // Creates a display filter with a name and an SQL expression.
+            let treesDisplayFilter = DisplayFilter(
+                name: "Trees",
+                whereClause: "req_type LIKE 'Tree Maintenance or Damage'"
+            )
+            
+            // Creates a display filter definition from the display filter.
+            let treesDisplayFilterDefinition = ManualDisplayFilterDefinition(
+                activeFilter: treesDisplayFilter,
+                availableFilters: [treesDisplayFilter]
+            )
+            return treesDisplayFilterDefinition
+        }()
+        
+        init() {
+            map.addOperationalLayer(featureLayer)
+        }
+        
+        /// Filters the feature layer based on a given filter mode.
+        /// - Parameter filterMode: The mode indicating how to filter the layer.
+        func filterFeatureLayer(filterMode: FilterMode) {
+            // Sets the feature layer's definition expression.
+            featureLayer.definitionExpression = filterMode == .definitionExpression
+            ? treesDefinitionExpression
+            : ""
+            
+            // Sets the feature layer's display filter definition.
+            featureLayer.displayFilterDefinition = filterMode == .displayFilterDefinition
+            ? treesDisplayFilterDefinition
+            : nil
```

Comment:
Consider using a switch statement, making this code much more readable:

```suggestion
            let (definitionExpression, displayFilterDefinition): (String, DisplayFilterDefinition?) = switch filterMode {
            case .definitionExpression:
                (treesDefinitionExpression, nil)
            case .displayFilterDefinition:
                ("", treesDisplayFilterDefinition)
            case .none:
                ("", nil)
            }
            // Sets the feature layer's definition expression.
            featureLayer.definitionExpression = definitionExpression
            // Sets the feature layer's display filter definition.
            featureLayer.displayFilterDefinition = displayFilterDefinition
```

Reply:
This also seems a little convoluted to me. Would it make sense to have two switch statements here? (One for each constant)

Reply:
No, there should definitely only be one switch statement. You could move the setting of the feature layer properties into the cases, which I originally started doing, but this way is less error prone (a case can't forget to set one of the properties).